devicetypes/smartthings/ikea-button.src/ikea-button.groovy

```diff
@@ -248,14 +280,18 @@ def parse(String description) {
 		log.debug "Creating event: ${event}"
 		sendEvent(event)
 	} else {
-		if ((description?.startsWith("catchall:")) || (description?.startsWith("read attr -"))) {
+		if (description?.startsWith('zone status')) {
+			event = parseIasMessage(description)
```

Comment:
this doesn't result in events for the child button being sent. you should probably be leveraging `sendButtonEvent` instead of your largely-redundant `getZoneButtonResult`

Reply:
Yeah, this should mostly be refactored so that all button handling logic is all inside `getButtonResult`